rpc-rs/src/chunkify.rs

```diff
@@ -100,66 +118,50 @@ impl ChunkEndpoint {
     }
 
     /// load response after de-chunking
-    pub fn get_unchunkified_response(&self, inv_id: &str) -> RpcResult<Vec<u8>> {
+    pub async fn get_unchunkified_response(&self, inv_id: &str) -> RpcResult<Vec<u8>> {
         // responses are stored in the object store with '-r' suffix on the object name
-        self.get_unchunkified(&format!("{}-r", inv_id))
+        self.get_unchunkified(&format!("{}-r", inv_id)).await
     }
 
     /// chunkify a message
     #[instrument(level = "trace", skip(self, bytes))]
-    pub fn chunkify(&self, inv_id: &str, bytes: &mut impl Read) -> RpcResult<()> {
-        let store = self.create_or_reuse_store()?;
+    pub async fn chunkify(
+        &self,
+        inv_id: &str,
+        mut bytes: (impl tokio::io::AsyncRead + std::marker::Unpin),
```

Comment:
Nit: This should either be imported up top so it can be shortened to `impl AsyncRead + Unpin` or to break it out as generic types instead

Reply:
Same nit goes for all of the places this happens in this file

Reply:
done